### DISCUSSION TITLE
[Question][Hw1] Testing Procedure Details

### DISCUSSION THREAD

**Discussion by anisrashidov** [prosyslab-classroom/cs348-information-security#687] [Q&A]
Name: Anis Rashidov

Could you explain me how _make test_ works?? I don't get the comparison it makes and it's output is a little confusing.

**Comment by KAIST-JongchanPark** [comment]
How make test works:
1. run your implementation for each testcases
2. save each stdout result into output files. (@@@.output)
3. get a diff between output and expected result (diff @@@.output @@@.expected)

**Comment by anisrashidov** [comment]
Could you be more specific please?
So what I do is I run _make test_ command and then I run individual commands for each functions like: _./aes enc test/key.txt test/iv.txt test/helloworld.txt_. But I don't get it either

**Comment by KAIST-JongchanPark** [comment]
Only executing command `make test` does automatically comparing diff between your output and expected result.

If you want to check each test case, following step would be better.
1. run build command `make`
2. run testcase command `./aes enc test/key.txt test/iv.txt test/helloworld.txt`
  2.1 if you want to save your stdout, use redirection. (i.e., `./aes enc test/key.txt test/iv.txt test/helloworld.txt > test/helloworld.output`
3. compare your output with expected result. `diff helloworld.output helloworld.expected`
  3.1. if you want to know exact byte value, use external tool like `xxd`

**Comment by KAIST-JongchanPark** [comment]
and please make your title with brief information about your question. It would be helpful for other students :D

**Comment by KihongHeo** [comment]
Please open if you have more questions.
